File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums) == 1:
            return 1
        
        high_value = nums[-1]
        idx = 1
        
        while idx < len(nums):
            if nums[idx] == nums[idx-1]:
                value = nums.pop(idx)
                nums.append(value)
                idx -= 1
            if nums[idx] == high_value:
                return idx+1
            idx += 1
            
[0,1,1,1,2,2,3,3,4,0]   

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    if(nums.length == 1) return 1
    let highNum = nums[nums.length - 1]
    for(let i = 1; i<nums.length; i++){
        if(nums[i] == nums[i-1]){
            if(nums[i] == highNum){
                return i
            }
            let value = nums.splice(i, 1)
            nums.push(value)
            i--
        }else if(nums[i] < nums[i-1]){
            return i
        }
    }
    return nums.length
};